js/core.js

```diff
@@ -666,6 +666,22 @@ var ERMrest = (function (module) {
             }
 
             return result;
+        },
+
+        // figure out if Table is pure and binary association table.
+        // binary: Has 2 outbound foreign keys.
+        // pure: There are no other columns apart from the foreign key columns and columns that are part of key and have serial4 type.
+        _isPureBinaryAssociation: function () {
```

Comment:
I don't see any logic here to verify that there is a composite key constraint combining all the columns from both foreign keys.